lua/nui/popup/border.lua

```diff
@@ -541,7 +544,7 @@ function Border:_relayout()
 end
 
 ---@param edge "'top'" | "'bottom'"
----@param text? nil | string | table # string or NuiText
+---@param text? nil | string | table | function # string or NuiText or render(width)
```

Comment:
```suggestion
---@param text? nil | string | table | function # string or NuiText or render(max_width, mid_char)
```

Also update the README for `NuiLine` and `Popup` about the changes?